packages/router/src/history.tsx

```diff
@@ -11,7 +11,10 @@ const createHistory = () => {
       return listenerId
     },
     navigate: (to: string) => {
-      global.history.pushState({}, '', to)
+      if (global?.location?.pathname !== to) {
+        global.history.pushState({}, '', to)
+      }
```

Comment:
Do we need to consider comparing things like `hash` and `search` in location?

Reply:
Hmm this is a good call. Let me take a stab at it we probably should handle this.